test/vaadin-dialog_draggable-resizable-test.html

```diff
@@ -574,5 +582,31 @@
         });
 
       });
+
+      // This also covers use-case of a overlay element (eg. combo-box) inside
+      // of a modal dialog
+      describe('modal and modeless dialog together', () => {
+        let modalDialog, modelessDialog;
+        beforeEach(() => {
+          const dialogs = fixture('bring-to-front');
+          modalDialog = dialogs[0];
+          modelessDialog = dialogs[1];
+        });
+
+        it('modal should not bring to front and close if a modeless dialog is on top', () => {
+          modalDialog.opened = true;
+          modelessDialog.opened = true;
+
+          const modalBackdrop = modalDialog.$.overlay.$.backdrop;
+
+          expect(modalDialog.$.overlay._last).to.be.false;
```

Comment:
nit: This test knows a bit too much about overlay's internals (the `_last` property). A black box test would be ideal: check that the expected overlay is the frontmost by using `elementFromPoint`